frontend/pages/DetailedViewPage.js

```diff
@@ -1,25 +1,35 @@
 import React, {Component} from 'react';
 import {searchByUPC} from "../utils/api";
-import { View, ActivityIndicator, StyleSheet} from 'react-native';
-
+import {View, ActivityIndicator, StyleSheet, FlatList} from 'react-native';
+import _ from 'lodash'
 import { Image, PricingCard, Overlay, Text, Input, Button } from 'react-native-elements';
 
-
 export default class DetailedViewPage extends Component {
-
     constructor(props) {
         super(props);
+        store = ''
+        //Had to wrap this in a try catch for when the component is loaded with no navigation (testing)
+        try {
+            store = this.props.navigation.state.params.store
+        } catch(err) {
+            console.log(err)
+        }
         this.state = {
             isLoading:true,
             priceOverlayVisible: false,
-            newPrice: ''
+            newPrice: '',
+            store: store
         }
     }
 
-
     async componentDidMount() {
         let data = await searchByUPC(this.props.navigation.state.params.upc);
-        this.setState({isLoading: false, ...data[0]})
+
+        //we only want the prices for the store we selected
+        oldPrices = _.find(data[0].stores, ['name', this.state.store]).prices
+
+        //add oldprices, as well as other item data, and let the app load
+        this.setState({isLoading: false, oldPrices: oldPrices, ...data[0]})
     }
 
     toggleChangePriceOverlay = () => {
```

Comment:
You should add JsDocs to each of these functions in your class. Especially the ones that you wrote that aren't just Overriding the ReactComponent functions (render, onMount, etc)

I would recommending reading this: http://usejsdoc.org/about-getting-started.html
It's very useful, especially when your code starts getting large enough that you need a reminder what functions do.

Reply:
Will go through and add JsDoc or other documentation